server/auth/types/basic/basic_auth.ts

```diff
@@ -119,7 +119,7 @@ export class BasicAuthentication extends AuthenticationType {
           request,
           this.coreSetup.http.basePath.serverBasePath
         );
-        const redirectLocation = `${this.coreSetup.http.basePath.serverBasePath}${LOGIN_PAGE_URI}?${nextUrlParam}`;
+        const redirectLocation = `${this.coreSetup.http.basePath.serverBasePath}${LOGIN_PAGE_URI_WITH_REDIRECT}`;
```

Comment:
will this redirect always to the home page instead of the one requested?

Reply:
changed back according after talking to @devardee